c7n/resources/s3.py

```diff
@@ -1465,26 +1465,22 @@ class SetPublicBlock(BucketActionBase):
                 resource: s3
                 filters:
                   - type: check-public-block
-                    scope: BlockPublicAcls
+                    scope: Any
                     enabled: False
                 actions:
                   - type: set-public-block
-                    # scope: <------ optional (All by default)
-                    #   - BlockPublicAcls
-                    #   - IgnorePublicAcls
-                    # state: enable <------ optional (enable by default)
+                    # BlockPublicAcls: true <------ All optional, true by default
     """
 
     schema = type_schema(
         'set-public-block',
-        scope={
-            'type': 'array',
-            'items': {'type': 'string',
-                     'enum': ['BlockPublicAcls', 'IgnorePublicAcls',
-                        'BlockPublicPolicy', 'RestrictPublicBuckets', 'All']}},
-        state={'type': 'string',
-               'enum': ['enable', 'disable']})
+        BlockPublicAcls={'type': 'boolean', 'default': True},
```

Comment:
So this changes the nature of this action.  Instead of setting only the fields specified, it now sets all of them all the time.  There's no preservation of any existing settings.  If you're just trying to set one field, you can't do that without potentially changing the other settings.

Reply:
I think it's logical to set all the settings if none are specified, since the action itself is "set-public-block".  you're intending to set something, so assuming you're setting all makes sense.  But if you instead specify only specific values, I think only those values should be set.  Others should use the existing value.

Reply:
Addressed in a4f1288. I totally agree here, on the last take I think I was being a bit too opinionated that it should always be turned on :) Thanks for the feedback, reflecting on the zen of it all and being more explicit has definitely helped to level up my craftsmanship.